lib/puppet/provider/user/windows_adsi.rb

```diff
@@ -124,7 +124,9 @@ def password
   end
 
   def password=(value)
-    # TODO: emit Puppet debug message here about setting a property that we can't verify when this is nil or ''
+    if @resource[:password].nil? || @resource[:password] == ''
+      Puppet.warning("Empty or blank passwords are not recommended. Empty or blank passwords may be allowed on your system based on policy, but Puppet cannot verify an empty or blank password with the operating system. Because of this a password change event will be executed by Puppet on every run.")
```

Comment:
Good catch @MosesMendoza - thanks.

Worth mentioning that in any commit messages as well.